doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -245,13 +250,21 @@ Requests to the environment
       Displays all assumptions and constants :n:`@qualid` relies on.
 
 
-.. cmd:: Search @qualid
+.. cmd:: Search @search_item {? {| {| inside | outside } {+ @qualid } | {+ @search_item } } }
+
+   .. insertprodn search_item search_item
+
+   .. prodn::
+      search_item ::= {? - } @string {? % @ident }
+      | {? - } @one_term
```

Comment:
My fear i that this sort of stuff is unwelcoming.  When I designed the rendering for the pattern language I was afraid that people would find it hard to read, or too complicated.  Having examples of different special cases helps not overwhelm the reader.

I guess my claim is that one large formula describing all of a tactic syntax is not a bad thing to have, but not necessarily a great entry point either.

Reply:
"a great entry point" - like in a tutorial?  I don't always read the syntax first, but it is visually distinctive and very easy to skip over.  The variants, less distinctive, because I think we have too many boxes in the HTML layout.  It looks cluttered.  I think boxes are OK for commands and tactics, since these begin subsections.  Emphasizing everything is in effect emphasizing nothing.  Maybe some of these could be bullet points.

![image](https://user-images.githubusercontent.com/1253341/77794021-6264d700-7028-11ea-890d-f9694d30dd8c.png)